src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {
+  FC, useEffect, useRef, useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo as TodoType } from '../../types/Todo';
+
+interface Props {
+  todo: TodoType,
+  deleteTodo: (todoId: number) => void,
+  isLoading: boolean,
+  isTemporary?: boolean,
+  updateTodo?: (updatedTodo: TodoType) => Promise<void>,
+  setLoadingTodoId?: (loadingTodoId: number[]) => void,
+  handleEditTodo?: (todoId: number, newTitle: string) => void,
+}
+
+export const Todo: FC<Props> = (props) => {
+  const {
+    todo,
+    deleteTodo,
+    isLoading,
+    isTemporary,
+    updateTodo,
+    setLoadingTodoId,
+    handleEditTodo,
+  } = props;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handleToggle = async () => {
+    setLoadingTodoId?.([todo.id]);
+
+    const newTodo = {
+      ...todo,
+      completed: !todo.completed,
+    };
+
+    await updateTodo?.(newTodo);
+    setLoadingTodoId?.([]);
+  };
+
+  const titleChanger = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedTitle(event.target.value);
+  };
+
+  const handleEditTitle = () => {
+    setIsEditing(true);
+
+    if (editedTitle === todo.title) {
+      setIsEditing(false);
+    }
```

Comment:
```suggestion
    if (editedTitle === todo.title) {
      setIsEditing(false);
      
      return;
    }
```
Don't forget to return, as in your case I still see a request and then an error